### PR TITLE
fix huber_distance error

### PR DESCRIPTION
**Pull request by zhechenyan**
This is related to issue #1309 . 
The bug has been fixed and related documents have been fixed as well.

### CODE REVIEW

File: dynet/nodes-similarities.cc [hunk 1]
```diff
@@ -94,7 +94,7 @@ void HuberDistance::forward_dev_impl(const MyDevice & dev, const vector<const Te
   Tensor diff(xs[0]->d, nullptr, xs[0]->device, xs[0]->mem_pool);
   diff.v = static_cast<float*>(scratch_allocator->allocate(diff.d.size() * sizeof(float)));
   tvec(diff).device(*dev.edevice) = tvec(*xs[0]) - tvec(*xs[1]);
-  t<0>(fx).device(*dev.edevice) = (tvec(diff).abs() < d).select(tvec(diff).square(), d * (2 * tvec(diff).abs() - d)).sum();
+  t<0>(fx).device(*dev.edevice) = (tvec(diff).abs() < d).select(0.5 * tvec(diff).square(), 0.5 * d * (2 * tvec(diff).abs() - d)).sum();
```
Comment (xunzhang) [inline dynet/nodes-similarities.cc]: `0.5 * d * (2 * tvec(diff).abs() - d)` is  not readable, `d * (tvec(diff).abs() - 0.5 * d)`?

**Comment by xunzhang** [comment]
Thanks!

**Comment by redpony** [comment]
Oops. Sorry about that :)

**Comment by xunzhang** [comment]
Forgot to modify the backward, I'll send a pr soon to fix this.

**Pull request merged**